thumbor/handlers/__init__.py

```diff
@@ -258,23 +269,22 @@ def _load_media(self, context):
 
         quality = self.context.request.quality
         if quality is None:
-            if image_extension == '.webp' and self.context.config.WEBP_QUALITY is not None:
+            if image_extension == '.webp' and \
+                            self.context.config.WEBP_QUALITY is not None:
                 quality = self.context.config.get('WEBP_QUALITY')
             else:
                 quality = self.context.config.QUALITY
 
-
-        media = None
-
         logger.debug('Reading image as {extension}'.format(
             extension=image_extension
         ))
 
         try:
             result = context.request.engine.read(image_extension, quality)
-            media = Media.from_result(result)
+            media = Media(result, mimetype=content_type)
```

Comment:
The `content_type` value needs to be re-evaluated from the `image_extension` or the `result` value, otherwise the engine could return an object that does not match the previously specified value.